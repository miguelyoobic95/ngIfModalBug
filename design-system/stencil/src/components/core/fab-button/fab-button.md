---
name: Fab Button
category: Core Components
---

A `yoo-fab-button` can be pressed to perform an action or to open other related actions inside a `yoo-fab-list`. A `yoo-fab-button` will scroll with the content if used in isolation but can be fixed to a specific location in the content by being used inside a `yoo-fab-container`.

An icon, title and handler fuction may be specified as a `FabButtonEntry`.

```typescript
    export interface FabButtonEntry {
        icon?: string;
        title?: string;
        handler?: Function;
    }
```

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`fabEntry`|FabButtonEntry|`{ }`   |an object containing the icon and/or title to show inside the fab button as well as a handler function|
|`text`|string||the text of the fab|
|`label`|string| |a label to attach next to the button inside the list|
|`disabled`|boolean|`false`|if true, the yoo-fab-button will be disable|
|`icon`|string||icon for the fab|
|`toggleActive`|fucniton|||
|`parentHasList`|boolean|`false`||
|`activated`|boolean|`false`||

```yoo-fab-button.html
    <yoo-fab-container top right>
        <yoo-fab-button class="accent"></yoo-fab-button>
    </yoo-fab-container>
    <yoo-fab-container top left>
        <yoo-fab-button class="danger"></yoo-fab-button>
    </yoo-fab-container>
    <yoo-fab-container bottom right>
        <yoo-fab-button disabled="true" class="warning"></yoo-fab-button>
    </yoo-fab-container>
    <yoo-fab-container bottom left>
        <yoo-fab-button class="success"></yoo-fab-button>
    </yoo-fab-container>
    <yoo-fab-container center middle>
        <yoo-fab-button class="gradient-danger"></yoo-fab-button>
    </yoo-fab-container>
```

```yoo-fab-button.js
 var btns = document.querySelectorAll('yoo-fab-button')
  for(var i =0; i < btns.length; i++){
    btns[i].fabEntry = {
      icon: 'yo-more',
      handler: () => console.log('calling handler')
    }
  }
```

```yoo-fab-button.css hidden
    .present-fabs {
        display: flex;
        justify-content: space-between;
        position: relative;
        width: 85%;
        height: 300px;
    }
```

## CSS

|Type|Name|Description|
|---|---|---|
|Large|`large`|yoo-fab-button large (56px)|
|Accent|`accent`|yoo-fab-button accent|
|Stable|`stable`|yoo-fab-button stable|
|Success|`success`|yoo-fab-button success|
|Danger|`danger`|yoo-fab-button danger|
|Info|`info`|yoo-fab-button info|
|Warning|`warning`|yoo-fab-button warning|
|Gradient Accent|`gradient-accent`|yoo-fab-button gradient-accent|
|Gradient Stable|`gradient-grey`|yoo-fab-button gradient-grey|
|Gradient Success|`gradient-success`|yoo-fab-button gradient-success|
|Gradient Danger|`gradient-danger`|yoo-fab-button gradient-danger|
|Gradient Info|`gradient-info`|yoo-fab-button gradient-info|
|Gradient Warning|`gradient-warning`|yoo-fab-button gradient-warning|

```yoo-fab-button-large.html
<div class="present-fabs">
        <yoo-fab-container top right>
            <yoo-fab-button class="large accent"></yoo-fab-button>
        </yoo-fab-container>
        <yoo-fab-container top left>
            <yoo-fab-button class="large danger"></yoo-fab-button>
        </yoo-fab-container>
        <yoo-fab-container bottom right>
            <yoo-fab-button class="large warning"></yoo-fab-button>
        </yoo-fab-container>
        <yoo-fab-container bottom left>
            <yoo-fab-button class="large success"></yoo-fab-button>
        </yoo-fab-container>
        <yoo-fab-container center middle>
            <yoo-fab-button class="gradient-danger"></yoo-fab-button>
        </yoo-fab-container>
</div>
```

```yoo-fab-button-large.js
 var btns = document.querySelectorAll('yoo-fab-button')
  for(var i =0; i < btns.length; i++){
    btns[i].fabEntry = {
      icon: 'yo-more',
      handler: () => console.log('calling handler')
    }
  }
```

```yoo-fab-button-large.css hidden
    .present-fabs {
        position: relative;
        display: flex;
        justify-content: space-between;
        width: 85%;
        height: 200px;
    }
```