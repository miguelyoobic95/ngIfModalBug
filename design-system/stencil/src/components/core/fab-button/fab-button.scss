

// margin between fab buttons inside a list
$fab-btn-margin: 0.5rem;
$mini-fab-btn-size: 2.1875rem;

@mixin fab-color($color) {
    .fab-button {
        yoo-button {
            /deep/.container {
                background: cssvar($color);
                .value {
                    color: cssvar(light);
                }
            }
        }
        yoo-badge {
            /deep/.outer-container {
                background: cssvar($color);
                .inner-container {
                    .inner-text {
                        color: cssvar(light);
                    }
                }
            }
        }
    }
}

@mixin class-fab-color-border($class,$color,$border-color){
    :host(.#{$class}) {
        @include fab-color($color);
        yoo-button {
            /deep/.container {
                border: 1px solid cssvar($border-color);
            }
        }
    }
}

:host-context(.top) {
    .fab-in-list.show {
        margin: 8px 0;
    }
}

:host-context(.bottom) {
    .fab-in-list.show {
        margin: 8px 0;
    }
}

:host-context(.left) {
    .fab-in-list.show {
        margin: 0 8px;
    }
}

:host-context(.right) {
    .fab-in-list.show {
        margin: 0 8px;
    }
}


:host {

    .fab-button {
        // position: absolute;
        color: cssvar(light);
        border-radius: 50%;

        yoo-button {
            /deep/.container {
                @include text-truncate();
            }
        }

    }

    .fab-in-list {
        position: relative;
        width: 2.625rem;
        height: 2.625rem;
        opacity: 0;
        visibility: hidden;
        transform: scale(0);

        yoo-badge {
            position: absolute;
            right: 3.4275rem;
            bottom: 0.625rem;
        }

        &.show {
            opacity: 1;
            visibility: visible;
            transform: scale(1);
        }

        &.mini {
            yoo-button {
                /deep/.container {
                    width: $mini-fab-btn-size;
                    height: $mini-fab-btn-size;
                    padding: 0.375rem 0rem;
                    
                }
            }
        }
        
    }
}


:host(.large) {
    .fab-button {
        yoo-button {
            /deep/.container {
                padding: 5px;
            }
        }
    }
}

:host(.top-right) {
    .fab-button {
        top: 5px;
        right: 5px;
    }
}

:host(.top-left) {
    .fab-button {
        top: 5px;
        left: 5px;
    }
}

:host(.bottom-right) {
    .fab-button {
        bottom: 5px;
        right: 5px;
    }
}

:host(.bottom-left) {
    .fab-button {
        bottom: 5px;
        left: 5px;
    }
}

// Colors - TO DO - add styles dynamically from inside the component so that the button SCSS is used
:host(.accent) {
    @include fab-color(accent);
}

:host(.dark) {
    yoo-button {
        /deep/.container {
            background-color: cssvar(dark);
            color: cssvar(dark);
            border: 1px solid cssvar(dark-110);
        }
    }
}
@include class-fab-color-border(success,success,success-110);
@include class-fab-color-border(danger,danger,danger-110);
@include class-fab-color-border(info,info,info-110);
@include class-fab-color-border(warning,warning,warning-110);
@include class-fab-color-border(gradient-accent,gradient-accent,accent-110);
@include class-fab-color-border(gradient-dark,gradient-dark,dark-110);
@include class-fab-color-border(gradient-success,gradient-success,success-110);
@include class-fab-color-border(gradient-info,gradient-info,info-110);
@include class-fab-color-border(gradient-danger,gradient-danger,danger-110);
@include class-fab-color-border(gradient-warning,gradient-warning,warning-110);
