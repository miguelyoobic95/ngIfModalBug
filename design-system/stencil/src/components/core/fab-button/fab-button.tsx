import { Component, Prop, Element, State } from '@stencil/core';

@Component({
    tag: 'yoo-fab-button',
    styleUrl: 'fab-button.scss',
    scoped: true
})
export class YooFabButtonComponent {

    @Prop() fabEntry: FabButtonEntry = {};
    @Prop() text: string;
    @Prop() icon: string;
    @Prop() disabled: boolean = false;
    @Prop() toggleActive: Function;
    @Prop() parentHasList: boolean = false;
    @Prop() activated: boolean = false;
    @Prop() label: string;

    // Toggle buttons in a list
    @State() inContainer: boolean = false;
    @State() inList: boolean = false;
    @State() activatedState: boolean = false;

     @Element() host: HTMLStencilElement;

    componentDidLoad() {
        const parentNode = this.host.parentElement;
        const parentTag = parentNode ? parentNode.nodeName : null;

        this.inContainer = (parentTag === 'YOO-FAB-CONTAINER');
        this.inList = (parentTag === 'YOO-FAB-LIST');
    }

    getButtonClasses(): CssClassMap {
        return {
            'fab-button': true,
            'fab-in-list': this.inList
        };
    }

    onClick(): void {
        if (this.inContainer && this.parentHasList) {
            this.toggleActive();
            this.activatedState = !this.activatedState;
        } else {
            if (this.fabEntry.handler && !this.disabled) {
                this.fabEntry.handler();
            }
        }
    }

    isActivated(): boolean {
        return (this.activatedState && this.inContainer);
    }

    renderListButton(): JSX.Element {
        return (
            <div class="list-button-container">
                {this.label ? <yoo-badge text={this.label}></yoo-badge> : ''}
                <yoo-button class="fab icon-only" disabled={this.disabled} icon={this.fabEntry.icon ? this.fabEntry.icon : (this.icon ? this.icon : '')} text={this.fabEntry.text ? this.fabEntry.text : (this.text ? this.text : '')}></yoo-button>
            </div>
        );
    }

    renderContainerButton(): JSX.Element {
        return (
            <yoo-button class="fab icon-only" disabled={this.disabled} icon={(this.isActivated() ? 'yo-close2' : (this.fabEntry.icon ? this.fabEntry.icon : (this.icon ? this.icon : '')))} text={(this.isActivated() ? '' : this.fabEntry.text ? this.fabEntry.text : this.text ? this.text : '' )}></yoo-button>
        );
    }
    render(): JSX.Element {
        const fabClasses = {
            ... this.getButtonClasses()
        };

        return (
            <div class={fabClasses} onClick={this.onClick.bind(this)}>
                {this.inList ? this.renderListButton() : this.renderContainerButton()}
            </div>
        );
    }
}
export interface FabButtonEntry {
    icon?: string;
    text?: string;
    handler?: Function;
}

export interface CssClassMap { [className: string]: boolean; }
