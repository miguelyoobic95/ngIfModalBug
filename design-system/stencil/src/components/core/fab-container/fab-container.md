---
name: Fab Container
category: Core Components
---

A `yoo-fab-container` is used to create `yoo-fab-button`s in a fixed absolute position. They may also contain `yoo-fab-list`s that contain buttons that are shown when the main `yoo-fab-button` in the `yoo-fab-container` is clicked. A single `yoo-fab-container` may hold multiple `yoo-fab-lists`.


## Attributes
|Name|Type|Default|Description|
|---|---|---|---|
|`animated`|boolean|   |determines if the fab container animates on animation|


```yoo-fab-container.html
    <yoo-fab-container top right>
        <yoo-fab-button class="accent"></yoo-fab-button>
    </yoo-fab-container>
    <yoo-fab-container top center>
        <yoo-fab-button class="gradient-accent"></yoo-fab-button>
    </yoo-fab-container>
    <yoo-fab-container top left>
        <yoo-fab-button class="danger"></yoo-fab-button>
    </yoo-fab-container>
    <yoo-fab-container left middle>
        <yoo-fab-button class="danger"></yoo-fab-button>
    </yoo-fab-container>
    <yoo-fab-container bottom right>
        <yoo-fab-button class="warning"></yoo-fab-button>
    </yoo-fab-container>
    <yoo-fab-container bottom left>
        <yoo-fab-button class="success"></yoo-fab-button>
    </yoo-fab-container>
    <yoo-fab-container center middle>
        <yoo-fab-button class="gradient-danger"></yoo-fab-button>
    </yoo-fab-container>
```

```yoo-fab-container.js
 var btns = document.querySelectorAll('yoo-fab-button')
  for(var i =0; i < btns.length; i++){
    btns[i].fabEntry = {
      icon: 'yo-more',
      handler: () => console.log('calling handler')
    }
  }
```