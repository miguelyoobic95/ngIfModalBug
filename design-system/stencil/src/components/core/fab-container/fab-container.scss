

// margin between the fab and the edge of the screen
$fab-margin: $padding-small;
$fab-size: 2.625rem;

:host {
    position: absolute;
    z-index: 1000;

    &[top] {
        top: $fab-margin;
    }

    &[bottom] {
        bottom: $fab-margin;
    }

    &[middle]{
        top: 50%;
        margin-top: -$fab-size / 2;
    }

    &[center]{
        left: 50%;
        margin-left: -$fab-size / 2  
    }    

    &[right] {
        right: $fab-margin;
    }

    &[left] {
        left: $fab-margin;
    }
}