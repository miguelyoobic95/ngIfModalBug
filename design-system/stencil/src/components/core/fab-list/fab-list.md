---
name: Fab List
category: Core Components
---

A `yoo-fab-list` is used inside a `yoo-fab-container` and it's children should only consist of `yoo-fab-button`'s and their respective properties.

## Attributes
|Name|Type|Default|Description|
|---|---|---|---|
|`side`|string|`top`|the side to which the list will be displayed|
|`mini`|boolean|`false`|reduce the size of all buttons inside the list|
|`activated`|boolean|`false`||
|`animated`|boolean|`false`|will animate fab button transition on activation|

### Side

`yoo-fab-lists` can be activated and opened onto any size.

```yoo-fab-list.html
<yoo-fab-container middle center animated="true">
    <yoo-fab-button id="btn1"></yoo-fab-button>
    <yoo-fab-list id="list1" side="left">
      <yoo-fab-button class="gradient-success"></yoo-fab-button>
      <yoo-fab-button class="info"></yoo-fab-button>
      <yoo-fab-button class="warning"></yoo-fab-button>
    </yoo-fab-list>
    <yoo-fab-list id="list2" side="right">
        <yoo-fab-button class="accent"></yoo-fab-button>
        <yoo-fab-button class="danger"></yoo-fab-button>
        <yoo-fab-button class="success"></yoo-fab-button>
    </yoo-fab-list>
    <yoo-fab-list id="list3" side="top">
        <yoo-fab-button label="Label1" class="accent"></yoo-fab-button></yoo-fab-button>
        <yoo-fab-button label="Label2" class="danger"></yoo-fab-button>
        <yoo-fab-button label="Label3" class="success"></yoo-fab-button>
    </yoo-fab-list>
    <yoo-fab-list id="list4" side="bottom">
        <yoo-fab-button label="Label1" class="accent"></yoo-fab-button>
        <yoo-fab-button label="Label2" class="danger"></yoo-fab-button>
        <yoo-fab-button label="Label3" class="success"></yoo-fab-button>
    </yoo-fab-list>
</yoo-fab-container>
```

```yoo-fab-list.js
var btn1 = document.querySelector('#btn1');
  btn1.fabEntry = {
    icon: 'yo-more'
  }

var list1 = document.querySelector('#list1');
  var btns1 = list1.querySelectorAll('yoo-fab-button')
  for(var i =0; i < btns1.length; i++){
    btns1[i].fabEntry = {
      icon: 'yo-more',
      handler: () => console.log('calling handler')
    }
  }

  var list2 = document.querySelector('#list2');
  var btns2 = list2.querySelectorAll('yoo-fab-button')
  for(var i =0; i < btns2.length; i++){
    btns2[i].fabEntry = {
      icon: 'yo-more',
      handler: () => console.log('calling handler')
    }
  }

 var list3 = document.querySelector('#list3');
  var btns3 = list3.querySelectorAll('yoo-fab-button')
  for(var i =0; i < btns3.length; i++){
    btns3[i].fabEntry = {
      icon: 'yo-more',
      handler: () => console.log('calling handler')
    }
  }

 var list4 = document.querySelector('#list4');
  var btns4 = list4.querySelectorAll('yoo-fab-button')
  for(var i =0; i < btns4.length; i++){
    btns4[i].fabEntry = {
      icon: 'yo-more',
      handler: () => console.log('calling handler')
    }
  }
```

### Mini

Buttons inside the list will be of smaller size than the main `yoo-fab-button`.

```yoo-fab-list-mini.html
<yoo-fab-container middle center animated="true">
    <yoo-fab-button id="btn1"></yoo-fab-button>
    <yoo-fab-list id="list1" side="left" mini="true">
      <yoo-fab-button class="gradient-success"></yoo-fab-button>
      <yoo-fab-button class="info"></yoo-fab-button>
      <yoo-fab-button class="warning"></yoo-fab-button>
    </yoo-fab-list>
    <yoo-fab-list id="list2" side="right" mini="true">
        <yoo-fab-button class="accent"></yoo-fab-button>
        <yoo-fab-button class="danger"></yoo-fab-button>
        <yoo-fab-button class="success"></yoo-fab-button>
    </yoo-fab-list>
    <yoo-fab-list id="list3" side="top" mini="true">
        <yoo-fab-button class="accent"></yoo-fab-button></yoo-fab-button>
        <yoo-fab-button class="danger"></yoo-fab-button>
        <yoo-fab-button class="success"></yoo-fab-button>
    </yoo-fab-list>
    <yoo-fab-list id="list4" side="bottom" mini="true">
        <yoo-fab-button class="accent"></yoo-fab-button>
        <yoo-fab-button class="danger"></yoo-fab-button>
        <yoo-fab-button class="success"></yoo-fab-button>
    </yoo-fab-list>
</yoo-fab-container>
```

```yoo-fab-list-mini.js
var btn1 = document.querySelector('#btn1');
  btn1.fabEntry = {
    icon: 'yo-fire'
  }

var list1 = document.querySelector('#list1');
  var btns1 = list1.querySelectorAll('yoo-fab-button')
  for(var i =0; i < btns1.length; i++){
    btns1[i].fabEntry = {
      icon: 'yo-more',
      handler: () => console.log('calling handler')
    }
  }

  var list2 = document.querySelector('#list2');
  var btns2 = list2.querySelectorAll('yoo-fab-button')
  for(var i =0; i < btns2.length; i++){
    btns2[i].fabEntry = {
      icon: 'yo-fire',
      handler: () => console.log('calling handler')
    }
  }

 var list3 = document.querySelector('#list3');
  var btns3 = list3.querySelectorAll('yoo-fab-button')
  for(var i =0; i < btns3.length; i++){
    btns3[i].fabEntry = {
      icon: 'yo-check',
      handler: () => console.log('calling handler')
    }
  }

 var list4 = document.querySelector('#list4');
  var btns4 = list4.querySelectorAll('yoo-fab-button')
  for(var i =0; i < btns4.length; i++){
    btns4[i].fabEntry = {
      icon: 'yo-fire',
      handler: () => console.log('calling handler')
    }
  }
```
