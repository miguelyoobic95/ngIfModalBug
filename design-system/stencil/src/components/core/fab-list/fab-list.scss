

$fab-btn-size: 2.625rem;

:host {
    position: absolute;
    min-height: $fab-btn-size;
    min-width:  $fab-btn-size;
    flex-direction: column;
    align-items: center;
    top: 0;

    &.right {
        @include flex;
        flex-direction: row;
        left: $fab-btn-size + 0.5rem;

        &[mini] {
            margin: 0.1875rem 0;
        }
    }

    &.left {
        @include flex;        
        flex-direction: row-reverse;
        right: $fab-btn-size + 0.5rem;

        &[mini] {
            margin: 0.1875rem 0;
        }
    }

    &.top {
        @include flex;        
        top: auto;
        bottom: $fab-btn-size + 0.5rem;
        flex-direction: column-reverse;

        &[mini] {
            margin: 0 0.25rem;            
        }
    }

    &.bottom {
        @include flex;
        top: $fab-btn-size + 0.5rem;

        &[mini] {
            margin: 0 0.25rem;            
        }
    }
}