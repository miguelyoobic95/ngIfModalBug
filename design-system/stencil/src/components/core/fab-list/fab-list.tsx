import { Component, Prop, Watch, Element } from '@stencil/core';
import { setAnimation } from '../../../utils/anim';

@Component({
    tag: 'yoo-fab-list',
    styleUrl: 'fab-list.scss',
    scoped: true
})
export class YooFabListComponent {

    @Prop() side: string = 'top';
    @Prop() mini: boolean = false;
    @Prop() activated: boolean = false;
    @Prop() animated: boolean = false;

    @Element() host: HTMLYooFabContainerElement;

    @Watch('activated')
    activatedChanged(activated: boolean): void {
        if (this.animated) {
            this.animatedDisplay(activated);
        } else {
            this.nonAnimatedDisplay(activated);
        }
    }

    animatedDisplay(activated: boolean): void {
        const yooBtns = this.host.querySelectorAll('.fab-button');
        const btnsArray = Array.from(Array(yooBtns.length).keys());
        const timeoutAnimated = 100;
        const buttonSize = 60;
        btnsArray.forEach(i => {
            setTimeout(() => {
                if (activated) {
                    yooBtns[i].classList.add('show');
                    setAnimation('fab', yooBtns[i], {distance: (buttonSize), direction: this.side, open: true});
                } else {
                    setAnimation('fab', yooBtns[i], {distance: (buttonSize), direction: this.side, open: false});
                    setTimeout (() => {
                        yooBtns[i].classList.remove('show');
                    }, 100);
                }
                if (this.mini) {
                    yooBtns[i].classList.add('mini');
                }
            }, (this.activated ? (timeoutAnimated * i) : (timeoutAnimated * (yooBtns.length - i))));
        });
    }

    nonAnimatedDisplay(activated: boolean) {
        const yooBtns = this.host.querySelectorAll('.fab-button');
        const btnsArray = Array.from(Array(yooBtns.length).keys());
        const timeoutNotAnimated = activated ? 30 : 0;
        btnsArray.forEach(i => {
            setTimeout(() => {
                activated ? yooBtns[i].classList.add('show') : yooBtns[i].classList.remove('show');
                if (this.mini) {
                    yooBtns[i].classList.add('mini');
                }
            }, timeoutNotAnimated * i);
        });
    }

    componentWillLoad() {
        this.host.classList.add(this.side);
    }

    render(): JSX.Element {
        return (
            <slot></slot>
        );
    }
}
