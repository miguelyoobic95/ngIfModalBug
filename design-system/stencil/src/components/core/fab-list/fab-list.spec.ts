import { TestWindow } from '@stencil/core/testing';
import { YooFabListComponent } from './fab-list';
import { YooFabContainerComponent } from '../fab-container/fab-container';
import { YooFabButtonComponent } from '../fab-button/fab-button';
jest.useFakeTimers();

describe('YooFabListComponent', () => {
    let window;
    beforeEach(() => {
        window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooFabListComponent()).toBeTruthy();
    });

    describe('rendering', () => {
        it('should render a yoo-fab-list with buttons inside', async () => {
            let element = await window.load({
                components: [YooFabListComponent, YooFabContainerComponent, YooFabButtonComponent],
                html: `<yoo-fab-container>
                    <yoo-fab-button></yoo-fab-button>
                    <yoo-fab-list id="list1" side="bottom">
                        <yoo-fab-button></yoo-fab-button>
                        <yoo-fab-button></yoo-fab-button>
                    </yoo-fab-list>
                </yoo-fab-container>`
             });
             expect(element).toMatchSnapshot();
             let lists = element.querySelectorAll('yoo-fab-list');
             expect(lists).toHaveLength(1);
             let list1 = element.querySelector('#list1');
             expect(list1.querySelectorAll('yoo-fab-button')).toHaveLength(2);
        });
        it('should render multiple yoo-fab-lists with buttons inside', async () => {
            let element = await window.load({
                components: [YooFabListComponent, YooFabContainerComponent, YooFabButtonComponent],
                html: `<yoo-fab-container>
                    <yoo-fab-button></yoo-fab-button>
                    <yoo-fab-list id="list1" side="bottom">
                        <yoo-fab-button></yoo-fab-button>
                    </yoo-fab-list>
                    <yoo-fab-list id="list2" side="bottom">
                        <yoo-fab-button></yoo-fab-button>
                        <yoo-fab-button></yoo-fab-button>
                        <yoo-fab-button></yoo-fab-button>
                    </yoo-fab-list>
                </yoo-fab-container>`
             });
             let lists = element.querySelectorAll('yoo-fab-list');
             expect(lists).toHaveLength(2);
             let list1 = element.querySelector('#list1');
             expect(list1.querySelectorAll('yoo-fab-button')).toHaveLength(1);
             let list2 = element.querySelector('#list2');
             expect(list2.querySelectorAll('yoo-fab-button')).toHaveLength(3);
             expect(element).toMatchSnapshot();
        });
    });

    describe('Opening and closing the list', () => {
        it('should activate the list and the main fab when the latter is pressed', async () => {
            let element = await window.load({
                components: [YooFabListComponent, YooFabContainerComponent, YooFabButtonComponent],
                html: `<yoo-fab-container>
                    <yoo-fab-button id="trigger"></yoo-fab-button>
                    <yoo-fab-list id="list1" side="right">
                        <yoo-fab-button></yoo-fab-button>
                        <yoo-fab-button></yoo-fab-button>
                    </yoo-fab-list>
                </yoo-fab-container>`
             });
            let triggerBtn = element.querySelectorAll('.fab-button')[0];
            await triggerBtn.click();
            await window.flush();
            expect(element.querySelector('yoo-fab-button').activated).toBeTruthy();
            expect(element.querySelector('yoo-fab-list').activated).toBeTruthy();
        });
        it('should add the show class to the buttons in the list', async () => {
            let element = await window.load({
                components: [YooFabListComponent, YooFabContainerComponent, YooFabButtonComponent],
                html: `<yoo-fab-container>
                    <yoo-fab-button id="trigger"></yoo-fab-button>
                    <yoo-fab-list id="list1" side="right">
                        <yoo-fab-button></yoo-fab-button>
                        <yoo-fab-button></yoo-fab-button>
                    </yoo-fab-list>
                </yoo-fab-container>`
             });
            expect(element.querySelectorAll('.fab-button.fab-in-list')).toHaveLength(2);
            let listContainer = element.querySelectorAll('yoo-fab-list')[0];
            listContainer.activated = true;
            jest.runOnlyPendingTimers();
            expect(element.querySelectorAll('.fab-button.fab-in-list.show')).toHaveLength(2);
        });
    });
});