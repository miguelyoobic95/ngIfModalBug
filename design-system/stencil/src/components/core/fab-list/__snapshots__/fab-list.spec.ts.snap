// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooFabListComponent rendering should render a yoo-fab-list with buttons inside 1`] = `
<yoo-fab-container
  class="hydrated"
  data-yoo-fab-container-host=""
  data-yoo-fab-container-slot=""
>
  
  
  
  
  
  
  
  
                    
  <yoo-fab-button
    class="hydrated"
    data-yoo-fab-button-host=""
  >
    
    <div
      class="fab-button"
      data-yoo-fab-button=""
    >
      <yoo-button
        class="fab icon-only"
        data-yoo-fab-button=""
        icon=""
        text=""
      />
    </div>
  </yoo-fab-button>
  
                    
  <yoo-fab-list
    class="bottom hydrated"
    data-yoo-fab-list-host=""
    data-yoo-fab-list-slot=""
    id="list1"
    side="bottom"
  >
    
    
    
    
    
    
    
    
                        
    <yoo-fab-button
      class="hydrated"
      data-yoo-fab-button-host=""
    >
      
      <div
        class="fab-button fab-in-list"
        data-yoo-fab-button=""
      >
        <div
          class="list-button-container"
          data-yoo-fab-button=""
        >
          
          <yoo-button
            class="fab icon-only"
            data-yoo-fab-button=""
            icon=""
            text=""
          />
        </div>
      </div>
    </yoo-fab-button>
    
                        
    <yoo-fab-button
      class="hydrated"
      data-yoo-fab-button-host=""
    >
      
      <div
        class="fab-button fab-in-list"
        data-yoo-fab-button=""
      >
        <div
          class="list-button-container"
          data-yoo-fab-button=""
        >
          
          <yoo-button
            class="fab icon-only"
            data-yoo-fab-button=""
            icon=""
            text=""
          />
        </div>
      </div>
    </yoo-fab-button>
    
                    
  </yoo-fab-list>
  
                
</yoo-fab-container>
`;

exports[`YooFabListComponent rendering should render multiple yoo-fab-lists with buttons inside 1`] = `
<yoo-fab-container
  class="hydrated"
  data-yoo-fab-container-host=""
  data-yoo-fab-container-slot=""
>
  
  
  
  
  
  
  
  
  
  
                    
  <yoo-fab-button
    class="hydrated"
    data-yoo-fab-button-host=""
  >
    
    <div
      class="fab-button"
      data-yoo-fab-button=""
    >
      <yoo-button
        class="fab icon-only"
        data-yoo-fab-button=""
        icon=""
        text=""
      />
    </div>
  </yoo-fab-button>
  
                    
  <yoo-fab-list
    class="bottom hydrated"
    data-yoo-fab-list-host=""
    data-yoo-fab-list-slot=""
    id="list1"
    side="bottom"
  >
    
    
    
    
    
    
                        
    <yoo-fab-button
      class="hydrated"
      data-yoo-fab-button-host=""
    >
      
      <div
        class="fab-button fab-in-list"
        data-yoo-fab-button=""
      >
        <div
          class="list-button-container"
          data-yoo-fab-button=""
        >
          
          <yoo-button
            class="fab icon-only"
            data-yoo-fab-button=""
            icon=""
            text=""
          />
        </div>
      </div>
    </yoo-fab-button>
    
                    
  </yoo-fab-list>
  
                    
  <yoo-fab-list
    class="bottom hydrated"
    data-yoo-fab-list-host=""
    data-yoo-fab-list-slot=""
    id="list2"
    side="bottom"
  >
    
    
    
    
    
    
    
    
    
    
                        
    <yoo-fab-button
      class="hydrated"
      data-yoo-fab-button-host=""
    >
      
      <div
        class="fab-button fab-in-list"
        data-yoo-fab-button=""
      >
        <div
          class="list-button-container"
          data-yoo-fab-button=""
        >
          
          <yoo-button
            class="fab icon-only"
            data-yoo-fab-button=""
            icon=""
            text=""
          />
        </div>
      </div>
    </yoo-fab-button>
    
                        
    <yoo-fab-button
      class="hydrated"
      data-yoo-fab-button-host=""
    >
      
      <div
        class="fab-button fab-in-list"
        data-yoo-fab-button=""
      >
        <div
          class="list-button-container"
          data-yoo-fab-button=""
        >
          
          <yoo-button
            class="fab icon-only"
            data-yoo-fab-button=""
            icon=""
            text=""
          />
        </div>
      </div>
    </yoo-fab-button>
    
                        
    <yoo-fab-button
      class="hydrated"
      data-yoo-fab-button-host=""
    >
      
      <div
        class="fab-button fab-in-list"
        data-yoo-fab-button=""
      >
        <div
          class="list-button-container"
          data-yoo-fab-button=""
        >
          
          <yoo-button
            class="fab icon-only"
            data-yoo-fab-button=""
            icon=""
            text=""
          />
        </div>
      </div>
    </yoo-fab-button>
    
                    
  </yoo-fab-list>
  
                
</yoo-fab-container>
`;
